src/edits.ts

```diff
@@ -199,6 +205,64 @@ async function uploadMappingFile(appEditId: string, versionCode: number, options
     }
 }
 
+async function uploadDebugSymbolsFile(appEditId: string, versionCode: number, options: EditOptions) {
+    if (options.debugSymbols != undefined && options.debugSymbols.length > 0) {
+        const fileStat = lstatSync(options.debugSymbols);
+
+        var data: Buffer | null = null;
+        if (fileStat.isDirectory()) {
+            data = await createDebugSymbolZipFile(options.debugSymbols);
+        }
+
+        if (data == null) {
+            data = readFileSync(options.debugSymbols);
+        }
+
+        if (data != null) {
+            core.debug(`[${appEditId}, versionCode=${versionCode}, packageName=${options.applicationId}]: Uploading Debug Symbols file @ ${options.debugSymbols}`);
+            const res = await androidPublisher.edits.deobfuscationfiles.upload({
+                auth: options.auth,
+                packageName: options.applicationId,
+                editId: appEditId,
+                apkVersionCode: versionCode,
+                deobfuscationFileType: 'nativeCode',
+                media: {
+                    mimeType: 'application/octet-stream',
+                    body: Readable.from(data)
+                }
+            })
+        }
+    }
+}
+
+async function createDebugSymbolZipFile(debugSymbolsPath: string) {
+    async function addDirectory(root: JSZip | null, dirPath: string, rootPath: string, isRootRoot: boolean) {
```

Comment:
Is there a reason a nested function was chosen here? Seems like something that's not really necessary, since there's only one usage that is called directly

Reply:
Yeah, a nested function isn't needed there. So I'll move that function up a level.